Tests/ApplicativeRouterTests/ApplicativeRouterTests.swift

```diff
@@ -12,21 +12,57 @@ func req(_ method: ApplicativeRouter.Method, _ location: String, _ body: Data? =
 
 class ApplicativeRouterTests: XCTestCase {
   func testRouter() {
+
     let router =
       Route.home <¢ .get <*| end
-        <|> Route.episode <¢> (.get <* lit("episode") *> .str) <*| end
-        <|> Route.episodes <¢ (.get <* lit("episodes")) <*| end
-        <|> Route.search <¢> (.get <* lit("search") *> opt(param("query"))) <*| end
-        <|> Route.signup <¢> (.post *> .dataBody) <* lit("signup") <*| end
+        // Route with path argument and two optional query params: /episode/:id?ref=foo&t=123
+        <|> curry(Route.episode) <¢> (.get <* lit("episode") *> .str) <*> opt(param("ref")) <*> opt(param("t", .int)) <*| end
+
+        // Route with no path arguments and two query params, `page` and `ref`. However, the `ref` param
+        // is parsed using the `refTag` helper that will provide a first class wrapper type for that value.
+        <|> Route.episodes <¢> refTag(.get <* lit("episodes") *> opt(param("page", .int))) <*| end
+
+        // Route with one query param: /search?query=foo
+        <|> Route.search <¢ .get <* lit("search") <*> opt(param("query")) <*| end
```

Comment:
I'm a lil confused why this is the correct way to express a route with a single query param. before, it was written as:

```swift 
Route.search <¢> (.get <* lit("search") *> opt(param("query"))) <*| end
```

and although that worked, it seemed counter-intuitive to have the `param` parser inside the parens where we do `.get <* lit` (lol). I wanted to bring it out of the parens so we could think of first parsing the path, and then the query params.

in order to do that, and get it in the form I now have, I had to change `<¢>` to `<¢`... can you help me understand that?

Reply:
oh nvm, I now get it. `Route.search <¢ .get` is executed first and we don't care about the result of the `.get` parser. nice!